File: STRINGS/2137/Main.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {

    static int[] getBooks(int n, Scanner scanner) {
        int[] books = new int[n];
        for (int i = 0; i < n; i++)
            books[i] = scanner.nextInt();
        Arrays.sort(books);
        return books;
    }

    static void formattedPrint(int[] books) {
        for (int book : books)
            System.out.printf("%04d\n", book);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNext()) {
            int n = scanner.nextInt();
            int[] books = getBooks(n, scanner);
            formattedPrint(books);
        }
        scanner.close();
    }
}

File: STRINGS/2137/main.rb
def get_books(n)
  books = []
  n.times { books.push(gets.to_i) }
  books.sort
end

def formatted_print(books)
  books.each { |book| puts format("%04d", book) }
end

while (n = gets&.to_i)
  books = get_books(n)
  formatted_print(books)
end
